### PR TITLE
mysql mysql-client 8.0.33

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mysql.rb [hunk 1]
```diff
@@ -184,3 +187,37 @@ def caveats
...
+   ENDIF()
++  MESSAGE(STATUS "ZLIB_FOUND = ${ZLIB_FOUND}")
+ ENDFUNCTION(FIND_SYSTEM_ZLIB)
+ 
+ MACRO (RESET_ZLIB_VARIABLES)
+@@ -114,9 +116,11 @@ MACRO (MYSQL_CHECK_ZLIB)
+     MYSQL_USE_BUNDLED_ZLIB()
+   ELSEIF(WITH_ZLIB STREQUAL "system")
+     FIND_SYSTEM_ZLIB()
++    MESSAGE(STATUS "ZLIB_FOUND = ${ZLIB_FOUND}")
```
Comment (carlocab) [inline Formula/mysql.rb:216]: So `ZLIB_FOUND` is true inside `FIND_SYSTEM_ZLIB` but not outside it? Odd.

**Comment by p-linnane** [comment]
I'm unable to figure out why linking with `libfido2` isn't working. Any ideas @Homebrew/core?

File: Formula/mysql-client.rb [hunk 2]
```diff
@@ -35,6 +35,10 @@ class MysqlClient < Formula
 
   fails_with gcc: "5"
 
+  # Fix for "Cannot find system zlib libraries" even though they are installed.
+  # https://bugs.mysql.com/bug.php?id=110745
+  patch :DATA
```
Comment (carlocab) [inline Formula/mysql-client.rb:40]: We should move this to Homebrew/formula-patches if we end up carrying this for another release.

**Review by carlocab — APPROVED** [review]
Weird re libfido, but ok.

File: Formula/mysql-client.rb (hunk 2 above)
Comment (p-linnane) [reply Formula/mysql-client.rb:40]: Upstream responded to that bug report today, so we should be ok:
>Fixed as of the upcoming MySQL Server 8.0.34 and 8.1.0 releases, and here's the proposed changelog entry from the documentation team:
>
>MySQL would not build with -DWITH_ZLIB=system; it'd complain about not
>finding the system zlib library despite finding it.
>
>Thank you for the bug report.

**Pull request merged**